app/Models/Post.php

```diff
@@ -41,13 +42,20 @@ class Post extends Model
     public function image(): Attribute
     {
         return Attribute::make(
-            get: fn (string $value) => filter_var($value, FILTER_VALIDATE_URL) ? $value : asset('storage/'.$value),
-            set: fn ($value): string => filter_var($value, FILTER_VALIDATE_URL) ? $value : $value->store('posts', 'public')
+            get: fn(string $value) => filter_var($value, FILTER_VALIDATE_URL) ? $value : asset('storage/' . $value),
+            set: fn($value): string => filter_var($value, FILTER_VALIDATE_URL) ? $value : $value->store('posts', 'public')
         );
     }
 
-    public function scopePublished($query)
+    public function scopePublished($query): Builder
     {
         return $query->where('published_at', '<=', now());
     }
+
+    public function scopeSearch($query, $search): Builder
+    {
+        return $query->when($search, function ($query, $search) {
+            return $query->where('title', 'like', '%' . $search . '%');
+        });
```

Comment:
Returns here do nothing since $query is an instance and updates its reference.

```suggestion
        $query->when($search, function (Builder $query, string $search) {
            $query->where('title', 'like', '%' . $search . '%');
        });
```